tools/scanner/src/analysis.rs

```diff
@@ -461,10 +494,23 @@ impl FnLoops {
 struct IteratorVisitor<'a> {
     props: FnLoops,
     body: &'a Body,
+    num_visited_blocks: usize,
 }
 
 impl<'a> MirVisitor for IteratorVisitor<'a> {
+    fn visit_basic_block(&mut self, bb: &BasicBlock) {
+        self.num_visited_blocks += 1;
+        self.super_basic_block(bb);
+    }
+
     fn visit_terminator(&mut self, term: &Terminator, location: Location) {
+        // A goto is identified as a loop latch if its target number is less than
+        // the number of visited basic block.
+        if let TerminatorKind::Goto { target } = &term.kind {
+            if self.num_visited_blocks > *target {
+                self.props.loops += 1;
+            }
+        }
```

Comment:
This is not true for MIR though, you need to traverse the graph.

Reply:
I changed it to find all goto with target that has been visited.

Reply:
I also add the visitor for other terminators that also contains targets beyond GOTO.